src/components/PopUp.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from "react";
+import { View, StyleSheet, ViewStyle, Modal, Alert } from "react-native";
+import type { StyleProp } from "react-native";
+import { PRIMARY_COLOR, TEXT_COLOR } from "../util/colors";
+import { Button } from "./Button";
+
+export interface ModalProps {
+	text: string;
+	accessibilityLabel: string;
+	accessibilityHint?: string;
+	style?: StyleProp<ViewStyle>;
+}
+
+export const PopUp: React.FC<ModalProps> = ({
+	text,
+	accessibilityLabel,
+	accessibilityHint,
+	style,
+	children,
+}) => {
+	const [modalVisible, setModalVisible] = useState(false);
+	return (
+		<View
+			accessibilityLabel="A Pop Up"
+			accessibilityHint="A Pop Up"
+			style={StyleSheet.compose(
+				{
+					backgroundColor: PRIMARY_COLOR,
+				},
+				style
+			)}
+		>
+			<Modal
+				animationType="none"
+				// TODO: setting 'transparent' to false will make the background disappear; we may want this for our users
+				// so that there are less distractions and they can focus on a single element. Test it out and lmk what you guys think
+				transparent={true}
+				visible={modalVisible}
+				onRequestClose={() => {
+					Alert.alert("Modal has been closed.");
+					setModalVisible(!modalVisible);
+				}}
```

Comment:
haven't been able to load the app because I'm on campus, but I don't think we should use words like "modal" with our users: those words are for us, and they probably won't understand what they mean. in fact, I don't think that an alert is even necessary when the dialogue box is closed.